core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -726,41 +699,24 @@ func (eb *EthBroadcaster) tryAgainWithNewEstimation(ctx context.Context, lgr log
 		logger.Sugared(eb.logger).AssumptionViolation(err.Error())
 		return err, false
 	}
-	keySpecificMaxGasPriceWei := eb.config.KeySpecificMaxGasPriceWei(etx.FromAddress)
-	gasPrice, gasLimit, err := eb.estimator.GetFee(ctx, etx.EncodedPayload, etx.GasLimit, keySpecificMaxGasPriceWei, txmgrtypes.OptForceRefetch)
-	if err != nil {
-		return errors.Wrap(err, "tryAgainWithNewEstimation failed to estimate gas"), true
-	}
-	lgr.Warnw("L2 rejected transaction due to incorrect fee, re-estimated and will try again",
-		"etxID", etx.ID, "err", err, "newGasPrice", gasPrice, "newGasLimit", gasLimit)
 
 	// TODO: nil handling for gasPrice.Legacy? will this ever be reached where EIP1559 is enabled on a L2 but a legacy tx?
-	return eb.tryAgainWithNewLegacyGas(ctx, lgr, etx, attempt, initialBroadcastAt, gasPrice.Legacy, gasLimit)
-}
-
-func (eb *EthBroadcaster) tryAgainWithNewLegacyGas(ctx context.Context, lgr logger.Logger, etx EthTx, attempt EthTxAttempt, initialBroadcastAt time.Time, newGasPrice *assets.Wei, newGasLimit uint32) (err error, retyrable bool) {
-	replacementAttempt, err := eb.NewLegacyAttempt(etx, newGasPrice, newGasLimit)
+	replacementAttempt, fee, feeLimit, retryable, err := eb.NewAttemptWithType(ctx, etx, lgr, attempt.TxType, txmgrtypes.OptForceRefetch)
```

Comment:
Whats the harm in calling the main NewAttempt() from here, ignoring setting the attempt.TxType?
I believe that function will correctly use the right TxType.
This way, we can delete this method variant from AttemptBuilder, making it more simpler.

Reply:
it's an L2 edge case where if an L2 implements EIP1559 then we would want to pass the information onward to make sure fee estimation is correct.

this would only be important in the transitionary period where there are still both legacy + dynamic txs in progress